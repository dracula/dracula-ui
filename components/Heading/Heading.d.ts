import React, { HTMLAttributes } from 'react';
import { MarginMixin, PaddingMixin } from '../../base/spacing';
export declare const headingSizes: {
    'heading-1': string;
    'heading-2': string;
    'heading-3': string;
    'heading-4': string;
    'heading-5': string;
    'heading-6': string;
};
export declare const headingColors: {
    purpleCyan: string;
    yellowPink: string;
    cyanGreen: string;
    pinkPurple: string;
    animated: string;
    cyan: string;
    green: string;
    orange: string;
    pink: string;
    purple: string;
    red: string;
    yellow: string;
    white: string;
    black: string;
    blackSecondary: string;
    blackLight: string;
};
/** Heading Props */
export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement>, MarginMixin, PaddingMixin {
    /**
     * The size and type of Heading to be used.
     * Denotes hierarchy.
     */
    size?: keyof typeof headingSizes;
    /**
     * The Dracula UI color to be applied to the Heading.
     */
    color?: keyof typeof headingColors;
    as?: keyof HTMLElementTagNameMap;
}
/**
 * Heading is used to display headlines and other forms of hierarchical Text.
 *
 * Headings are similar to the base Text component, but restricted to certain
 * sizes and font weight.
 */
export declare const Heading: React.FC<HeadingProps>;
