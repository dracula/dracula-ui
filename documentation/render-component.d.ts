import React from 'react';
export declare type ComponentExample = {
    html: string;
    docs: string;
    title: string;
    react?: any;
    screenshot?: string;
    svgData?: string;
};
export declare type SnapshotBuilder = {
    comp: () => React.ReactElement;
    docs: string;
    title: string;
};
export declare function snapshot(title: string, comp: () => React.ReactElement, docs?: string): SnapshotBuilder;
export declare function renderSnapshot(title: string, comp: () => React.ReactElement, docs?: string): ComponentExample;
