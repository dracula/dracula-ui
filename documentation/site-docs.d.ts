import { SnapshotBuilder } from './render-component';
import React from 'react';
interface Documentation {
    basic: () => SnapshotBuilder;
    variations: () => Array<SnapshotBuilder> | SnapshotBuilder;
}
export declare function docs<T>(componentClass: React.FunctionComponent<T> | string, documentation: Documentation): void;
export {};
