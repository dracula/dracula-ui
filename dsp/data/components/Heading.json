{
  "class": "component",
  "type": "page",
  "id": "Heading",
  "name": "Heading",
  "last_updated": "2021-03-14T22:34:30.073Z",
  "last_updated_by": "System",
  "description": "Heading is used to display headlines and other forms of hierarchical Text.\n\nHeadings are similar to the base Text component, but restricted to certain\nsizes and font weight.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "heading",
    "languages": {
      "javascriptreact": "<Heading>Heading</Heading>;\n",
      "html": "<h1\n  class=\"drac-heading drac-heading-1 drac-text-white drac-spacing-none drac-margin-none\"\n>\n  Heading\n</h1>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<h1\n  class=\"drac-heading drac-heading-1 drac-text-white drac-spacing-none drac-margin-none\"\n>\n  Heading\n</h1>\n",
      "react": "<Heading>Heading</Heading>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Heading.png",
      "svgData": "./dsp/assets/svgs/Heading.svg"
    },
    {
      "title": "Sizes",
      "html": "<div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"white\"\n      class=\"drac-heading drac-heading-1 drac-text-white drac-spacing-none drac-margin-none\"\n    >\n      heading-1\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h2\n      color=\"white\"\n      class=\"drac-heading drac-heading-2 drac-text-white drac-spacing-none drac-margin-none\"\n    >\n      heading-2\n    </h2>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h3\n      color=\"white\"\n      class=\"drac-heading drac-heading-3 drac-text-white drac-spacing-none drac-margin-none\"\n    >\n      heading-3\n    </h3>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h4\n      color=\"white\"\n      class=\"drac-heading drac-heading-4 drac-text-white drac-spacing-none drac-margin-none\"\n    >\n      heading-4\n    </h4>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h5\n      color=\"white\"\n      class=\"drac-heading drac-heading-5 drac-text-white drac-spacing-none drac-margin-none\"\n    >\n      heading-5\n    </h5>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h6\n      color=\"white\"\n      class=\"drac-heading drac-heading-6 drac-text-white drac-spacing-none drac-margin-none\"\n    >\n      heading-6\n    </h6>\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-1\">\n      heading-1\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-2\">\n      heading-2\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-3\">\n      heading-3\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-4\">\n      heading-4\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-5\">\n      heading-5\n    </Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\" size=\"heading-6\">\n      heading-6\n    </Heading>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/HeadingSizes.png",
      "svgData": "./dsp/assets/svgs/HeadingSizes.svg"
    },
    {
      "title": "Colors",
      "html": "<div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"white\"\n      class=\"drac-heading drac-heading-1 drac-text-white drac-spacing-none drac-margin-none\"\n    >\n      white\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"black\"\n      class=\"drac-heading drac-heading-1 drac-text-black drac-spacing-none drac-margin-none\"\n    >\n      black\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"blackSecondary\"\n      class=\"drac-heading drac-heading-1 drac-text-black-secondary drac-spacing-none drac-margin-none\"\n    >\n      blackSecondary\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"blackLight\"\n      class=\"drac-heading drac-heading-1 drac-text-black-light drac-spacing-none drac-margin-none\"\n    >\n      blackLight\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"cyan\"\n      class=\"drac-heading drac-heading-1 drac-text-cyan drac-spacing-none drac-margin-none\"\n    >\n      cyan\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"green\"\n      class=\"drac-heading drac-heading-1 drac-text-green drac-spacing-none drac-margin-none\"\n    >\n      green\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"orange\"\n      class=\"drac-heading drac-heading-1 drac-text-orange drac-spacing-none drac-margin-none\"\n    >\n      orange\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"pink\"\n      class=\"drac-heading drac-heading-1 drac-text-pink drac-spacing-none drac-margin-none\"\n    >\n      pink\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"purple\"\n      class=\"drac-heading drac-heading-1 drac-text-purple drac-spacing-none drac-margin-none\"\n    >\n      purple\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"red\"\n      class=\"drac-heading drac-heading-1 drac-text-red drac-spacing-none drac-margin-none\"\n    >\n      red\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"yellow\"\n      class=\"drac-heading drac-heading-1 drac-text-yellow drac-spacing-none drac-margin-none\"\n    >\n      yellow\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"purpleCyan\"\n      class=\"drac-heading drac-heading-1 drac-text-purple-cyan drac-spacing-none drac-margin-none\"\n    >\n      purpleCyan\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"yellowPink\"\n      class=\"drac-heading drac-heading-1 drac-text-yellow-pink drac-spacing-none drac-margin-none\"\n    >\n      yellowPink\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"cyanGreen\"\n      class=\"drac-heading drac-heading-1 drac-text-cyan-green drac-spacing-none drac-margin-none\"\n    >\n      cyanGreen\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"pinkPurple\"\n      class=\"drac-heading drac-heading-1 drac-text-pink-purple drac-spacing-none drac-margin-none\"\n    >\n      pinkPurple\n    </h1>\n  </div>\n  <div style=\"margin-bottom: 20px\">\n    <h1\n      color=\"animated\"\n      class=\"drac-heading drac-heading-1 drac-text-animated drac-spacing-none drac-margin-none\"\n    >\n      animated\n    </h1>\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"white\">white</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"black\">black</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"blackSecondary\">blackSecondary</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"blackLight\">blackLight</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"cyan\">cyan</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"green\">green</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"orange\">orange</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"pink\">pink</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"purple\">purple</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"red\">red</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"yellow\">yellow</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"purpleCyan\">purpleCyan</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"yellowPink\">yellowPink</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"cyanGreen\">cyanGreen</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"pinkPurple\">pinkPurple</Heading>\n  </div>\n  <div style={{ marginBottom: 20 }}>\n    <Heading color=\"animated\">animated</Heading>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/HeadingColors.png",
      "svgData": "./dsp/assets/svgs/HeadingColors.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "size": {
      "defaultValue": null,
      "description": "The size and type of Heading to be used.\nDenotes hierarchy.",
      "name": "size",
      "parent": {
        "fileName": "src/components/Heading/Heading.tsx",
        "name": "HeadingProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        }
      ],
      "required": false,
      "type": {
        "name": "\"heading-1\" | \"heading-2\" | \"heading-3\" | \"heading-4\" | \"heading-5\" | \"heading-6\" | undefined"
      }
    },
    "color": {
      "defaultValue": null,
      "description": "The Dracula UI color to be applied to the Heading.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Heading/Heading.tsx",
        "name": "HeadingProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "spacing": {
      "defaultValue": null,
      "description": "Controls the spacing between the Heading component and its parent and siblings.",
      "name": "spacing",
      "parent": {
        "fileName": "src/components/Heading/Heading.tsx",
        "name": "HeadingProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        }
      ],
      "required": false,
      "type": { "name": "SpacingPropType | undefined" }
    },
    "margin": {
      "defaultValue": null,
      "description": "Dracula UI standard margin properties.",
      "name": "margin",
      "parent": {
        "fileName": "src/components/Heading/Heading.tsx",
        "name": "HeadingProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        }
      ],
      "required": false,
      "type": { "name": "MarginPropType | undefined" }
    },
    "as": {
      "defaultValue": null,
      "description": "",
      "name": "as",
      "parent": {
        "fileName": "src/components/Heading/Heading.tsx",
        "name": "HeadingProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        }
      ],
      "required": false,
      "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Heading is used to display headlines and other forms of hierarchical Text.\n\nHeadings are similar to the base Text component, but restricted to certain\nsizes and font weight.",
    "displayName": "Heading",
    "methods": [],
    "props": {
      "size": {
        "defaultValue": null,
        "description": "The size and type of Heading to be used.\nDenotes hierarchy.",
        "name": "size",
        "parent": {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Heading/Heading.tsx",
            "name": "HeadingProps"
          }
        ],
        "required": false,
        "type": {
          "name": "\"heading-1\" | \"heading-2\" | \"heading-3\" | \"heading-4\" | \"heading-5\" | \"heading-6\" | undefined"
        }
      },
      "color": {
        "defaultValue": null,
        "description": "The Dracula UI color to be applied to the Heading.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Heading/Heading.tsx",
            "name": "HeadingProps"
          }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "spacing": {
        "defaultValue": null,
        "description": "Controls the spacing between the Heading component and its parent and siblings.",
        "name": "spacing",
        "parent": {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Heading/Heading.tsx",
            "name": "HeadingProps"
          }
        ],
        "required": false,
        "type": { "name": "SpacingPropType | undefined" }
      },
      "margin": {
        "defaultValue": null,
        "description": "Dracula UI standard margin properties.",
        "name": "margin",
        "parent": {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Heading/Heading.tsx",
            "name": "HeadingProps"
          }
        ],
        "required": false,
        "type": { "name": "MarginPropType | undefined" }
      },
      "as": {
        "defaultValue": null,
        "description": "",
        "name": "as",
        "parent": {
          "fileName": "src/components/Heading/Heading.tsx",
          "name": "HeadingProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Heading/Heading.tsx",
            "name": "HeadingProps"
          }
        ],
        "required": false,
        "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
      }
    }
  }
}
