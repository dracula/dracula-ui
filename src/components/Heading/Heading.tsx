import cx from 'classnames/dedupe'
import { mapValues } from 'lodash'
import React, { HTMLAttributes } from 'react'
import { colors } from '../../base/colors'
import {
  cleanProps,
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'

export const headingSizes = {
  'heading-1': 'drac-heading-1',
  'heading-2': 'drac-heading-2',
  'heading-3': 'drac-heading-3',
  'heading-4': 'drac-heading-4',
  'heading-5': 'drac-heading-5',
  'heading-6': 'drac-heading-6'
}

export const headingColors = mapValues(colors, (className) => {
  return className.replace('-bg-', '-text-')
})

/** Heading Props */
export interface HeadingProps
  extends HTMLAttributes<HTMLHeadingElement>,
    MarginMixin,
    PaddingMixin {
  /**
   * The size and type of Heading to be used.
   * Denotes hierarchy.
   */
  size?: keyof typeof headingSizes

  /**
   * The Dracula UI color to be applied to the Heading.
   */
  color?: keyof typeof headingColors

  as?: keyof HTMLElementTagNameMap
}

/**
 * Heading is used to display headlines and other forms of hierarchical Text.
 *
 * Headings are similar to the base Text component, but restricted to certain
 * sizes and font weight.
 */
export const Heading: React.FC<HeadingProps> = (props: HeadingProps) => {
  const { size = 'heading-1', color, as, ...htmlProps } = props
  const tag = {
    'heading-1': 'h1',
    'heading-2': 'h2',
    'heading-3': 'h3',
    'heading-4': 'h4',
    'heading-5': 'h5',
    'heading-6': 'h6'
  }

  const finalProps = {
    ...htmlProps,
    className: cx(
      `drac-heading`,
      props.className,
      headingSizes[size ?? 'heading-1'],
      headingColors[color ?? 'white'],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  return React.createElement(
    as ?? tag[size],
    cleanProps(finalProps),
    props.children
  )
}

Heading.displayName = 'Heading'
