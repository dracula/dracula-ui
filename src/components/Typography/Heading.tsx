import cx from 'classnames'
import { mapValues } from 'lodash'
import React, { HTMLAttributes } from 'react'
import { colors } from '../../base/colors'
import { spacingClasses, SpacingPropType } from '../../base/spacing'

export const headingSizes = {
  'heading-1': 'drac-heading-1',
  'heading-2': 'drac-heading-2',
  'heading-3': 'drac-heading-3',
  'heading-4': 'drac-heading-4',
  'heading-5': 'drac-heading-5',
  'heading-6': 'drac-heading-6'
}

export const headingColors = mapValues(colors, (className) => {
  return className.replace('-bg-', '-text-')
})

/** Heading Props */
export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
  /**
   * The size and type of Heading to be used.
   * Denotes hierarchy.
   */
  size?: keyof typeof headingSizes

  /**
   * The Dracula UI color to be applied to the Heading.
   */
  color?: keyof typeof headingColors

  /**
   * Controls the spacing between the Heading component and its parent and siblings.
   */
  spacing?: SpacingPropType
}

/**
 * Heading is used to display headlines and other forms of hierarchical Text.
 *
 * Headings are similar to the base Text component, but restricted to certain
 * sizes and font weight.
 */
export const Heading: React.FC<HeadingProps> = (props: HeadingProps) => {
  const tag = {
    'heading-1': 'h1',
    'heading-2': 'h2',
    'heading-3': 'h3',
    'heading-4': 'h4',
    'heading-5': 'h5',
    'heading-6': 'h6'
  }

  const size = tag[props.size ?? 'heading-1']
  const finalProps = {
    ...props,
    className: cx(
      `drac-heading`,
      headingSizes[props.size ?? 'heading-1'],
      headingColors[props.color ?? 'white'],
      ...spacingClasses(props.spacing ?? 'none')
    )
  }

  return React.createElement(size, finalProps, props.children)
}

Heading.displayName = 'Heading'
