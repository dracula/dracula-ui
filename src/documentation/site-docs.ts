import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from './render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from './pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'
import os from 'os'

interface Documentation {
  basic: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder> | SnapshotBuilder
}

jest.setTimeout(10000)

const RECORDING = process.env.RECORDING

export function docs<T>(
  componentClass: React.FunctionComponent<T> | string,
  documentation: Documentation
) {
  const name =
    typeof componentClass === 'string'
      ? componentClass
      : componentClass.displayName ?? componentClass.name

  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeAll(async () => {
      if (!RECORDING) {
        return
      }

      let executablePath = undefined
      if (os.arch() === 'arm64') {
        executablePath = '/Applications/Chromium.app/Contents/MacOS/Chromium'
      }

      browser = await puppeteer.launch({
        headless: true,
        devtools: true,
        executablePath
      })
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      if (!RECORDING) {
        return
      }

      await browser.close()
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basic()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      if (RECORDING) {
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          null,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg
      }

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations: SnapshotBuilder[] = ([] as SnapshotBuilder[]).concat(
      documentation.variations()
    )

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        if (RECORDING) {
          const [screenshot, svg] = await componentScreenshot(
            browser,
            snapshot,
            variation,
            name
          )

          snapshot.screenshot = screenshot
          snapshot.svgData = svg
        }

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
