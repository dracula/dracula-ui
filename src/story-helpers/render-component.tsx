import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { pretty } from './pretty'

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
  screenshot?: string
  svgData?: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): ComponentExample {
  let component = comp()
  const html = mount(component).html()
  let wrapped = false

  // wrap the component in a div when it's not wrapped already
  // we do this so we can get the actual component example to not be shallow rendered.
  // Shallow rendering a parent "div" gives us a nice JSX representation of the example.
  if (component.type !== 'div') {
    component = <div>{component}</div>
    wrapped = true
  }

  let react = shallow(component).debug({ verbose: true })

  if (wrapped) {
    // replace wrapper div ðŸ˜¬
    react = react
      .replace(/^<div>\n/, '')
      .replace(/\n<\/div>$/, '')
      .trim()
  }

  return {
    title,
    html: pretty(html, 'html'),
    react: pretty(react, 'typescript'),
    docs
  }
}
