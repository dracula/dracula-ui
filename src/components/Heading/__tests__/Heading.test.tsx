import { docs } from '@/documentation/site-docs'
import React from 'react'
import { snapshot } from '../../../documentation/render-component'
import { Heading } from '../Heading'

docs(Heading, {
  basic() {
    return snapshot(
      'Usage',
      () => <Heading>Heading</Heading>,
      'Heading is an abstraction/style that can/is applied to HTML heading elements.'
    )
  },
  variations() {
    return [
      snapshot(
        'Sizes',
        () => (
          <div>
            <Heading size="2xl">2xl</Heading>
            <Heading size="xl">xl</Heading>
            <Heading size="lg">lg</Heading>
            <Heading size="md">md</Heading>
            <Heading size="sm">sm</Heading>
            <Heading size="xs">xs</Heading>
          </div>
        ),
        'Heading sizes range from xs to 2xl using the Dracula UI font size scale.'
      ),
      snapshot(
        'Tags',
        () => (
          <div>
            <Heading as="h1">h1</Heading>
            <Heading as="h2">h2</Heading>
            <Heading as="h3">h3</Heading>
            <Heading as="h4">h4</Heading>
            <Heading as="h5">h5</Heading>
            <Heading as="h6">h6</Heading>
          </div>
        ),
        'By default, every Heading is specified with `h2`, but you can change that to better match your SEO needs.'
      ),
      snapshot(
        'Colors',
        () => (
          <div>
            <Heading size="2xl" color="pink">
              pink
            </Heading>
            <Heading size="2xl" color="purpleCyan">
              purpleCyan
            </Heading>
          </div>
        ),
        'Headings can be customized to use any of the Dracula UI theme colors, including all solid, gradient, and animated colors.'
      )
    ]
  }
})
