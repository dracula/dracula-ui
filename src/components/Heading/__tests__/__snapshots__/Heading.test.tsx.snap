// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Heading Basic Usage 1`] = `
Object {
  "docs": "Heading is an abstraction/style that can/is applied to HTML heading elements.",
  "html": "<h1 class=\\"drac-heading drac-heading-1 drac-text-white\\">Heading</h1>
",
  "react": "<Heading>Heading</Heading>;
",
  "title": "Usage",
}
`;

exports[`Site: Heading Colors 1`] = `
Object {
  "docs": "Headings can be customized to use any of the Dracula UI theme colors, including all solid, gradient, and animated colors.",
  "html": "<div>
  <h1 class=\\"drac-heading drac-heading-1 drac-text-pink\\">pink</h1>
  <h1 class=\\"drac-heading drac-heading-1 drac-text-purple-cyan\\">purpleCyan</h1>
</div>
",
  "react": "<div>
  <Heading size=\\"heading-1\\" color=\\"pink\\">
    pink
  </Heading>
  <Heading size=\\"heading-1\\" color=\\"purpleCyan\\">
    purpleCyan
  </Heading>
</div>;
",
  "title": "Colors",
}
`;

exports[`Site: Heading Sizes 1`] = `
Object {
  "docs": "Each heading size is mapped directly to a specific h tag in HTML.",
  "html": "<div>
  <h1 class=\\"drac-heading drac-heading-1 drac-text-white\\">heading-1</h1>
  <h2 class=\\"drac-heading drac-heading-2 drac-text-white\\">heading-2</h2>
  <h3 class=\\"drac-heading drac-heading-3 drac-text-white\\">heading-3</h3>
  <h4 class=\\"drac-heading drac-heading-4 drac-text-white\\">heading-4</h4>
  <h5 class=\\"drac-heading drac-heading-5 drac-text-white\\">heading-5</h5>
  <h6 class=\\"drac-heading drac-heading-6 drac-text-white\\">heading-6</h6>
</div>
",
  "react": "<div>
  <Heading size=\\"heading-1\\">heading-1</Heading>
  <Heading size=\\"heading-2\\">heading-2</Heading>
  <Heading size=\\"heading-3\\">heading-3</Heading>
  <Heading size=\\"heading-4\\">heading-4</Heading>
  <Heading size=\\"heading-5\\">heading-5</Heading>
  <Heading size=\\"heading-6\\">heading-6</Heading>
</div>;
",
  "title": "Sizes",
}
`;
